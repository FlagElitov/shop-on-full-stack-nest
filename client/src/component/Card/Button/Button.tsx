import React from "react";
import Button from "@material-ui/core/Button";

interface ButtonComponentProps {
  id: number;
  toggleHandler: (id: number) => void;
}

const ButtonComponent: React.FC<ButtonComponentProps> = ({
  id,
  toggleHandler,
}) => {
  return (
    <Button
      variant="contained"
      onClick={() => toggleHandler(id)}
      color="primary"
    >
      Отправить 100$
    </Button>
  );
};

export default ButtonComponent;
