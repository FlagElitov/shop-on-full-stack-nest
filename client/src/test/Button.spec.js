import { shallow } from "enzyme";
import React from "react";
import Button from "../component/Card/Button/Button";

const setUp = () => shallow(<Button />);

describe("Unit Testing App component", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });
  it("App Snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
