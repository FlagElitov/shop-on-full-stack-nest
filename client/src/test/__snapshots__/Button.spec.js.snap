// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing App component App Snapshot 1`] = `
<WithStyles(ForwardRef(Button))
  color="primary"
  onClick={[Function]}
  variant="contained"
>
  Отправить 100$
</WithStyles(ForwardRef(Button))>
`;
